action/audit.go

```diff
@@ -19,29 +26,49 @@ func (s *Action) Audit(c *cli.Context) error {
 	dupes := make(map[string][]string)
 	foundWeakPasswords := false
 
-	for _, secret := range t.List(0) {
-		content, err := s.Store.Get(secret)
+	pwList := t.List(0)
+	fmt.Printf("Checking %d secrets. This may take some time ...\n", len(pwList))
+	bar := pb.StartNew(len(pwList))
+	for _, secret := range pwList {
+		bar.Increment()
+		content, err := s.Store.GetFirstLine(secret)
 		if err != nil {
-			return err
+			fmt.Println(color.RedString("Failed to retrieve secret '%s': %s", secret, err))
+			continue
 		}
 
 		pw := string(content)
+		if len(pw) > maxCheckLength {
+			fmt.Println(color.RedString("Warning: Skipped long secret"))
```

Comment:
Might be something we could even do upstream in crunchy. Dictionary lookups with that length certainly make no sense.